### ISSUE TITLE
Spatial mechanics improvements

### ISSUE THREAD

**Issue by uhloin** [AppliedEnergistics/Applied-Energistics-2#6642]
### Describe the feature

May be it is rarely used in the past but now it is must-have feature to move flawless blocks to home and create farm of cristals and dust. Therefore I propose some improvements for enjoj playing. Thank you again for a cool mod and for your constant efforts to improve it!

1. Better documentation
1.1. Make in wiki separate chapter for the spatial stuff.
1.2. Add description for all spatial cells. Currently I found separate page for 2x2x2 and 128x128x128 and no pages for 16x16x16.
1.3. Place into wiki the image of minimal spatial containment: I/O port + battery + 6 pylons + button. Players should get enough info from wiki instead of search videos.
1.4. The Patchouli is not used by AE2 but can at least URL to wiki page be displayed in tooltips or in interfaces? For example for spatial I/O port?

2. Better look
2.1. Make some cool animation of translocation blocks into port or back. At least those violet lightnings as for grow accelerators.
2.2. Make "button" inside the I/O port to activate it (and keep possibility of use redstone). It is more nice than sneak+stick button to the port and press it.

3. Better usability
3.1. Write in tooltip how many blocks are stored in spatial cell or at least is it empty or filled. 
Now it shows tooltip like "Serial... Stored size 1x1x1" for once used cell and "Unformatted. Capacity 2x2x2" for never used.
Should be "Stored size 1x1x1, 1 block(s)" for filled and "Empty size 2x2x2" for deployed.
3.2. If "Serial..." has some meaning it should be described in wiki.

4. Better cost balance
4.1. Allow "reformat" cell. For example if it has size 1x1x2 but containment is 1x1x1 then allow change size to 1x1x1. Such cells are too expensive to create a new for every x-y-z combination.
4.2. For start of game make cheap and tiny 1x1x1 cell and cheap "unsafe" I/O port only for such cells. It should store block and deploy with 100% success but after deploy has 50% chance to burn out and 25% chance burn out the cell inside it.

### Reasons why it should be considered

Very useful but hard-to-understand feature

**Comment by Wraithypooo** [comment]
I don't think that the spatial setups are harder to use than the rest of AE2. It's a mindset, and once you're in it, everything seems quite logic (according to the great book of Wraithypooo) :)
But it's really annoying that as soon as you go past copying a single block, not only do you format the cell to that amount (or less) but also to the north/south/east/west direction of said blocks. reformatting the cells is a good option, but only half the story.

**Comment by SirOlrik** [comment]
I second on the reformatting
And unconditional horizontal rotation and mirroring.

A way to empty content in to a normal storage, using the I/O Port.

Use normal memeory cells (1k, 4k, 16k, 32k ...), just like you do with the Fluids.

No swapping, only placemnet:
If I have a Pylon setup for a 1x1x5 area.
Placing the items in the world.
I have 2 additional blocks I wanted to put in front.
Used the same 16^3 cell.
Loaded the remaining 2 blocks (1x1x2) into the cell.
Place the Pylon setup, 2 blocks in front.
Placing the items in the world.
What the Pylon setup did, was exchanging the 1x1x5 area, so I ended up withe the 2 blocks in correct place, but the following 3 blocks where moved to the cell.
In other words, it considers empty space, as being a part of the structure.
I didn't recon that, I was expecting it to empty it's buffer, no more, no less.

**Comment by Sea-Kerman** [comment]
I'm about to write the guidebook page on spatial storage, so the documentation issue should go away.

The reason it swaps (and possibly why you can't reformat it) is because it actually works a lot like compact machines, creating a physical box in the spatial storage dimension where stuff is moved to and from. There is no "buffer". This is inherently incompatible with using normal cells and emptying them into normal storage, what if there's a block with no direct drop, or with some specific blockstate stuff where it doesn't map to an item directly?

**Comment by SirOlrik** [comment]
That makes sence :)

**Comment by uhloin** [comment]
> Wraithypooo

The problem is not a complexity of spatial storage itself but the logic of gameplay evolution.
1. The start of AE2 gameplay is mining of certus.
2. The running from meteorite to main base is not interesting, it's boring. So you need flawless block as soon as possible.
3. To move flawless block to your base you must use spatial things.
4. But for spatial things you need to kill many Endermans, it's not a start a game but a bit later. And even more you need go to Nether for Glowing Dust and Nether Quartz. It's already middle of game.

So my point is AE2 need to change a bit gameplay sequence. For example, allow to mine flawless block with iron pickaxe without silk touch. Or put flawless block not in every meteorite. Or set a reasonable chance that flawless block stay flawless after broken. If you want add some hardness (but possible for player at begin of game) just put meteorite core near to bedrock level under crater or make craters very rare thing to go far (with compass).

The main idea is a sequence. At start of game you need 1) certus farm for 2) some storage better than pile of chests.
So ideally you should be able to achieve 1) and 2) at start of game without Nether materials and ender pearls.

**Comment by Sea-Kerman** [comment]
It is perfectly possible to fully automate a certus farm with non-flawless budding blocks, which will repair and replace the budding block.

**Comment by SirOlrik** [comment]
I agree with [Sea-Kerman](https://github.com/Sea-Kerman) on tihs one.
It's fully automateable.
As a fact, in the last setup I had. I just made a bunch of Flawed Budding Cerus Quartz, and ended quickly up with loads of Certus Quartsn Crystals.
And then comes the fun of moving the Flawless version and finally make a neat setup for automation.

In other words: Something to do in both start and mid .game, rather than rushing through the game.
